lib/Server.js

```diff
@@ -213,6 +213,18 @@ class Server {
           ...options.client.overlay,
         };
       }
+
+      // Respect infrastructureLogging.level
+      if (typeof options.client.logging === "undefined") {
+        const configWithInfrastructureLogging =
+          configArray.find(
+            (config) =>
+              config.infrastructureLogging && config.infrastructureLogging.level
+          ) || configArray[0];
+
+        options.client.logging =
+          configWithInfrastructureLogging.infrastructureLogging.level;
+      }
```

Comment:
I think we need better logic here, ideally we should get options from compiler where we found `devServer` options